fuzz/fuzz_targets/int_set_op_processor.rs

```diff
@@ -0,0 +1,1045 @@
+        self.0 = min(self.0 + 1, Self::MAX_VALUE);
+    }
+}
+
+impl Domain<SmallInt> for SmallInt {
+    fn to_u32(&self) -> u32 {
+        self.0
+    }
+
+    fn from_u32(member: InDomain) -> SmallInt {
+        SmallInt::new(member.value())
+    }
+
+    fn is_continous() -> bool {
+        true
+    }
+
+    fn ordered_values() -> impl DoubleEndedIterator<Item = u32> {
+        0..=Self::MAX_VALUE
+    }
+
+    fn ordered_values_range(
+        range: RangeInclusive<SmallInt>,
+    ) -> impl DoubleEndedIterator<Item = u32> {
+        if range.start().0 > Self::MAX_VALUE || range.end().0 > Self::MAX_VALUE {
+            panic!("Invalid range of the SmallInt set.");
+        }
+        range.start().to_u32()..=range.end().to_u32()
+    }
+
+    fn count() -> usize {
+        Self::MAX_VALUE as usize + 1
+    }
+}
+
+/// This is an even integer in the domain of [0, 2048). It's used by the fuzzer
+/// for testing inverted sets + discontinous domains.
+#[derive(Debug, Copy, Clone, PartialEq, PartialOrd, Eq, Ord)]
+pub struct SmallEvenInt(u32);
+
+impl SmallEvenInt {
+    const MAX_VALUE: u32 = 4 * 512 - 2;
+
+    fn new(value: u32) -> SmallEvenInt {
+        if value > Self::MAX_VALUE {
+            panic!("Constructed SmallEvenInt with value > MAX_VALUE.");
+        }
+        if value % 2 != 0 {
+            panic!("Constructed SmallEvenInt with an odd value.");
+        }
+        SmallEvenInt(value)
+    }
+}
+
+impl SetMember<SmallEvenInt> for SmallEvenInt {
+    fn create(val: u32) -> Option<SmallEvenInt> {
+        if val > Self::MAX_VALUE || val % 2 != 0 {
+            return None;
+        }
+        Some(SmallEvenInt::new(val))
+    }
+
+    fn can_be_inverted() -> bool {
+        true
+    }
+
+    fn increment(&mut self) {
+        self.0 = min(self.0 + 2, Self::MAX_VALUE);
+    }
+}
+
+impl Domain<SmallEvenInt> for SmallEvenInt {
+    fn to_u32(&self) -> u32 {
+        self.0
+    }
+
+    fn from_u32(member: InDomain) -> SmallEvenInt {
+        SmallEvenInt::new(member.value())
+    }
+
+    fn is_continous() -> bool {
+        false
+    }
+
+    fn ordered_values() -> impl DoubleEndedIterator<Item = u32> {
+        (0..=(Self::MAX_VALUE / 2)).map(|ord| ord * 2)
+    }
+
+    fn ordered_values_range(
+        range: RangeInclusive<SmallEvenInt>,
+    ) -> impl DoubleEndedIterator<Item = u32> {
+        if range.start().0 > Self::MAX_VALUE || range.end().0 > Self::MAX_VALUE {
+            panic!("Invalid range of the SmallInt set.");
+        }
+        ((range.start().to_u32() / 2)..=(range.end().to_u32() / 2)).map(|ord| ord * 2)
+    }
+
+    fn count() -> usize {
+        ((Self::MAX_VALUE / 2) + 1) as usize
+    }
+}
+
+struct Input<'a, T>
+where
+    T: SetMember<T>,
+{
+    // The state includes 2 of each type of sets to allow us to test out binary set operations (eg. union)
+    int_set: &'a mut IntSet<T>,
+    btree_set: &'a mut BTreeSet<T>,
+}
+
+impl<T> Input<'_, T>
+where
+    T: SetMember<T>,
```

Comment:
Also just a style thing but generally when there's only a single (or a small number) of generic bounds it's easier to express them inline, like:

```rust
impl<T: SetMember<T>> Input<'_, T> { .. }
```